clean_air/data/storage.py

```diff
@@ -198,3 +181,211 @@ def put(self, item: AURNSite) -> None:
+
+    def _to_cached_file_path(self, dataset_id: str) -> Path:
+        return (self._cache_dir / Path(dataset_id) / Path(f"{dataset_id}.metadata")).absolute()
+
+    def get(self, dataset_id: str) -> Metadata:
+        metadata_key = self._to_s3_key(dataset_id)
+        cached_metadata_file = self._to_cached_file_path(dataset_id)
+
+        try:
+            self._fs.get(metadata_key, str(cached_metadata_file))
+        except PermissionError:
+            msg = f"PermissionError when accessing {metadata_key}."
+            msg += " Object may not exist, or you may have incorrect/misconfigured credentials"
+            raise DataStoreException(msg)
+
+        # Load Metadata
+        with cached_metadata_file.open() as md_file:
+            return self._metadata_serialiser.deserialise(md_file.read())
+
+    def put(self, item: Metadata) -> None:
+        if self._fs.anon:
+            raise DataStoreException("Cannot perform write operations in anonymous mode. "
+                                     "Please provide credentials with write permissions")
+
+        with tempfile.NamedTemporaryFile() as tmp_metadata:
+            tmp_metadata.write(self._metadata_serialiser.serialise(item).encode("utf-8"))
+            tmp_metadata.flush()
+            key = self._to_s3_key(item.id)
+            LOGGER.debug(f"Uploading metadata from {tmp_metadata.name} to {key}")
+            try:
+                self._fs.put(tmp_metadata.name, key)
+            except PermissionError:
+                raise DataStoreException(
+                    f"You do not have permission to upload to s3://{key}."
+                    " Please check your credentials are correct or contact the system administrator")
+
+
+def create_aurn_datastore(
+        bucket_name: str = "aurn", data_file_path: str = "AURN_Site_Information.csv",
+        endpoint_url: str = JasminEndpointUrls.EXTERNAL,
+        anon: bool = True) -> AURNSiteDataStore:
+    """
+    Return an AURNSiteDatastore instance configured with the given information
+
+    :param bucket_name: Name of the bucket storing the AURN site data CSV file
+    :param data_file_path: Object key of the AURN site data CSV file
+    :param endpoint_url: the object store service endpoint URL. Changes depending on whether accessing data from inside
+        or outside JASMIN, or using data stored on another AWS S3 compatible object store
+    :param anon: Whether to use anonymous access or credentials. anon=True is required for write access
+    """
+
+    s3_args = {"endpoint_url": endpoint_url}
+    if anon:
+        s3_args["config"] = Config(signature_version=UNSIGNED)
+
+    s3 = boto3.resource("s3", **s3_args)
+    bucket = s3.Bucket(bucket_name)
+    data_file_obj = bucket.Object(data_file_path)
+    try:
+        s3.meta.client.head_bucket(Bucket=bucket.name)
+        # data_file_obj.load()  # This fails, not clear why, permissions? It uses Head Object API call under the hood
```

Comment:
Is this a comment from a previous section waiting to be removed, or is it here as a reminder maybe?

Reply:
sorta.
It's the answer to the question "why are we only checking access to the Bucket?"